src/lib/y2users/user.rb

```diff
@@ -103,7 +101,10 @@ def full_name
     def clone_to(config)
       attrs = ATTRS.each_with_object({}) { |a, r| r[a] = public_send(a) }
       attrs.delete(:name) # name is separate argument
-      self.class.new(config, name, attrs)
+      cloned = self.class.new(config, name, attrs)
+      cloned.password = password.clone_to(config) if password
```

Comment:
Should not it be `password.clone_to(user)`? I think we don't need to assign a password to a config anymore. Otherwise, we can end with passwords assigned to an user that does not belong to the config of the password.

Reply:
:thinking: :thinking: Good catch!

Reply:
well, it is for consistency. For each object it is important under which context it lives ( aka system, staging, etc ). Of course it can be user and then transitent config as user lives in that context, but I prefer consistency. It e.g. allows easier changes in some staging graf. If you modify user you simply create new one with new parameters and replace old one and just assign password to that user if it is same.
Situation that password belong to user with different config is of course issue, but it should be easily detected and is same issue as if user has group that does not exist in given config.

Reply:
Yes, I see your point about having objects under a specific context. But for the password, I think the user is who defines the context. Maybe we can do something like:

~~~
cloned_user.password = password.duplicate
~~~

What do you think?

Reply:
yes, as said on call it is possible to have password live just as reference in user and clone it with simple `cloned_user.password = password.clone`